src/net.h

```diff
@@ -421,19 +422,19 @@ struct CombinerAll
     }
 };
 
-// Signals for message handling
-struct CNodeSignals
+// Interface for message handling
+class NetEventsInterface
 {
-    boost::signals2::signal<bool (CNode*, CConnman&, std::atomic<bool>&), CombinerAll> ProcessMessages;
-    boost::signals2::signal<bool (CNode*, CConnman&, std::atomic<bool>&), CombinerAll> SendMessages;
-    boost::signals2::signal<void (CNode*, CConnman&)> InitializeNode;
-    boost::signals2::signal<void (NodeId, bool&)> FinalizeNode;
+protected:
+    NetEventsInterface() = default;
+public:
+    virtual ~NetEventsInterface() = default;
```

Comment:
To make sure that the base gets cleaned up if, for example, it's stored as an abstract std::unique_ptr<NetEventsInterface>